### PR TITLE
Maintain only one list of tests

### PR DESCRIPTION
**Pull request by df7cb**
Make evaluates "=" variables lazily, so we can add items to TESTS later in the file and still have them show up in REGRESS.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -104,17 +100,17 @@ healpix_bare/healpix_bare.o : healpix_bare/healpix_bare.c
 pg_version := $(word 2,$(shell $(PG_CONFIG) --version))
 has_support_functions = $(if $(filter-out 9.% 10.% 11.%,$(pg_version)),y,n)
 
-crushtest: REGRESS += $(CRUSH_TESTS)
+crushtest: TESTS += $(CRUSH_TESTS)
```
Comment (esabol) [inline Makefile:103]: This change leaves out `init` from `crushtest`, right? Why would we want to do that?

Comment (df7cb) [reply Makefile:103]: crushtest is unaffected by this change, it runs the normal "installcheck" target, just with more tests. The list of tests is in $(TESTS) now, $(REGRESS) is only used internally by the pgxs rules that invoke pg_regress.

Adding to REGRESS would also work for this use case, but all the other rules in the Makefile need to add to TESTS so "make test" is also supported, so I think it's cleaner to have crushtest append to TESTS as well.

Comment (esabol) [reply Makefile:103]: Sorry, where is the `installcheck` target defined? I can't find it.

Comment (df7cb) [reply Makefile:103]: In PostgreSQL's pgxs.mk. The tests are run on `$(REGRESS)`, our Makefile defines `REGRESS     = init $(TESTS)`, crushtest appends to `TESTS`, so when `crushtest` is run, it includes the crush files.

Comment (esabol) [reply Makefile:103]: Well, OK then!

**Review by esabol — APPROVED** [review]

**Review by vitcpp — APPROVED** [review]

**Comment by df7cb** [comment]
@vitcpp I can rebase the PRs in whatever order you want to merge

**Comment by vitcpp** [comment]
@df7cb ok, thanks. Lets start from the simpler PRs then.

**Pull request merged**